Sources/App/Controllers/UserController.swift

```diff
@@ -44,19 +44,25 @@ class UserController: RouteCollection {
         }
     }
     
-    func login(_ req: Request, _ loginRequest: LoginRequest) throws -> Future<HTTPStatus> {
+    func login(_ req: Request, _ loginRequest: LoginRequest) throws -> Future<HTTPResponse> {
         let userService = try req.make(UserService.self)
         
         return userService.authorize(
             email: loginRequest.email,
             password: loginRequest.password,
             on: req
-        ).map { user in
+        )
+        .map(to: PublicUserResponse.self) { user in
             guard let user = user else {
-                return .unauthorized
+                throw BasicValidationError("Could not authorize User")
```

Comment:
We should return here 401 Unauthorized status code, `BasicValidationError` will return `400 Bad Request` here. I'd suggest reverting this one.

Reply:
Reverted, but got
```
error: type 'PublicUserResponse' has no member 'unauthorized'
                return .unauthorized
                       ~^~~~~~~~~~~~
```

Switched to `throw Abort(...)`